src/control/server/mgmt_svc.go

```diff
@@ -197,6 +206,15 @@ func (svc *mgmtSvc) Join(ctx context.Context, req *mgmtpb.JoinReq) (*mgmtpb.Join
 		return nil, errors.Wrap(err, "unmarshal Join response")
 	}
 
+	// if join successful, record membership
+	if resp.GetStatus() == 0 && resp.GetState() == mgmtpb.JoinResp_IN {
+		svc.members = append(svc.members, &SystemMember{
```

Comment:
Should this append be protected by a mutex? It's possible that multiple ms clients will invoke this RPC concurrently, right? I don't know this area of the code very well but there appears to be precedent for synchronized access to things here.